### PR TITLE
Command that listens to CrawlFinishedMessage

### PR DESCRIPTION
**Pull request by asturcon**
Closes the issue https://github.com/gbif/pipelines/issues/66

Steps followed to test it:
- Created a `crawler` exchange in /users/mlopez vh
- Run the propagator command in the IDE to be able to debug it. Used these params:
`xmlToAvro --conf avro-propagator.yaml`
- Published a message in the RabbitMQ UI with routing key `crawl.finished` and the following payload: 
```
{ "datasetUuid":"ea33eca1-c704-4062-a111-98f7b323e824", "attempt":"1", "totalRecordCount":"1" , "reason":"NORMAL"}
```
- Checked that the `AvroPropagatorCallback` is called. 

NOTE: `avro-propagator.yaml `contains:
```
messaging:
host: mq.gbif.org
virtualHost: /users/mlopez
username: ****
password: ****
queueName: xml-avro-propagator
threadCount: 1```

### CODE REVIEW

File: crawler-cli/src/main/java/org/gbif/crawler/xml/propagater/AvroPropagatorCommand.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.gbif.crawler.xml.propagater;
+
+import org.gbif.cli.Command;
+import org.gbif.cli.service.ServiceCommand;
+
+import com.google.common.util.concurrent.Service;
+import org.kohsuke.MetaInfServices;
+
+@MetaInfServices(Command.class)
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/xml/propagater/AvroPropagatorCommand.java]: Missing javadoc

File: crawler-cli/src/main/java/org/gbif/crawler/xml/propagater/AvroPropagatorConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.gbif.common.messaging.config.MessagingConfiguration;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParametersDelegate;
+import com.google.common.base.Objects;
+
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/xml/propagater/AvroPropagatorConfiguration.java]: javadoc

File: crawler-cli/src/main/java/org/gbif/crawler/xml/propagater/AvroPropagatorService.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  protected void shutDown() throws Exception {
+    listener.close();
+  }
+
+  private static class AvroPropagatorCallback extends AbstractMessageCallback<CrawlFinishedMessage> {
+
+    @Override
+    public void handleMessage(CrawlFinishedMessage message) {
+
+      LOG.info("CrawlFinsihedMessage recevied");
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/xml/propagater/AvroPropagatorService.java]: spelling mistakes

File: crawler-cli/src/main/resources/avro-propagator.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+messaging:
```
Comment (fmendezh) [inline crawler-cli/src/main/resources/avro-propagator.yaml]: remove this file

**Comment by asturcon** [comment]
Done with the changes

File: crawler-cli/src/main/java/org/gbif/crawler/pipelines/xml/XmlToAvroService.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  protected void shutDown() throws Exception {
+    listener.close();
+  }
+
+  private static class AvroPropagatorCallback extends AbstractMessageCallback<CrawlFinishedMessage> {
+
+    @Override
+    public void handleMessage(CrawlFinishedMessage message) {
+
+      LOG.info("CrawlFinsihedMessage received");
```
Comment (fmendezh) [inline crawler-cli/src/main/java/org/gbif/crawler/pipelines/xml/XmlToAvroService.java]: spelling mistake: CrawlFinishedMessage

**Pull request merged**